src/relayclient/GsnTest.ts

```diff
@@ -0,0 +1,57 @@
+import Web3 from 'web3'
+
+import CommandsLogic, { DeploymentResult } from '../cli/CommandsLogic'
+import { GsnDevProvider } from './GsnDevProvider'
+import { configureGSN } from './GSNConfigurator'
+import { DevGSNConfig } from './DevRelayClient'
+import { networks } from '../cli/utils'
+
+interface TestEnvironment {
+  deployment: DeploymentResult
+  devProvider: GsnDevProvider
+}
+
+class GsnTestClass {
+  private fixture?: TestEnvironment
+
+  /**
+   *
+   * @param host
+   * @param paymaster TODO: will allow using custom paymaster (need to provide ABI file contents)
+   * @return
+   */
+  async start (host?: string, paymaster?: any): Promise<void> {
```

Comment:
I like to call this pattern `init()`: its a workaround  for Javascript's deficiency of constructor, which can't be asynchrounous. 
so you create a class with `GsnTest gsntest = await new GsnTest(...).init()`
(with/without params to init)

BTW: in this specific case, you can actually do factory function, since you don't have constructor params at all..)

Reply:
Used subset of suggested API in your e-mail.